README.md

```diff
@@ -411,45 +412,102 @@ You can also use the option `--config-path` on the main command to set another c
+```yaml
+# Required, otherwise ggshield considers the file to use the deprecated v1 format
+version: 2
+
+# Set to true if the desired exit code for the CLI is always 0, otherwise the
+# exit code will be 1 if incidents are found.
+exit-zero: false # default: false
+
+verbose: false # default: false
+
+instance: https://api.gitguardian.com # default: https://api.gitguardian.com
+
+# Maximum commits to scan in a hook.
+max-commits-for-hook: 50 # default: 50
+
+# Accept self-signed certificates for the API.
+allow-self-signed: false # default: false
+
+secret:
+  # Exclude files and paths by globbing
+  ignored-paths:
+    - '**/README.md'
+    - 'doc/*'
+    - 'LICENSE'
+
+  # Ignore security incidents with the SHA256 of the occurrence obtained at output or the secret itself
+  ignored-matches:
+    - name:
+      match: 530e5a4a7ea00814db8845dd0cae5efaa4b974a3ce1c76d0384ba715248a5dc1
+    - name: credentials
+      match: MY_TEST_CREDENTIAL
+
+  show-secrets: false # default: false
+
+  # Detectors to ignore.
+  ignored-detectors: # default: []
+    - Generic Password
+```
+
+## Updating a v1 configuration file
+
+To update a v1 configuration file to a v2 configuration file, follow these steps:
+
+1. Add a `version: 2` entry.
+2. If the configuration file contains an `all-policies` key, remove it: it's no longer supported.
+3. If the configuration file contains an `ignore-default-excludes` key, remove it: it's no longer supported.
+4. If the configuration file contains an `api-url` key, replace it with an `instance` key, pointing to the _dashboard_ URL.
+5. If the configuration file contains one of the following keys: `paths-ignore`, `matches-ignore`, `show-secrets`, `banlisted-detectors`:
+   1. Create a `secret` key.
+   2. Move `paths-ignore` to `secret.ignored-paths`.
+   3. Move `matches-ignore` to `secret.ignored-matches`. If some match entries are strings instead of (`name`, `match`) objects, turn them into (`name`, `match`) objects.
+   4. Move `banlisted-detectors` to `secret.ignored-detectors`.
+   5. Move `show-secrets` to `secret.show-secrets`.
+
+Here is an example of a v1 configuration file:
+
+```yaml
+all-policies: false
+
+api-url: https://example.com/exposed
+
+show-secrets: true
+
 paths-ignore:
   - '**/README.md'
-  - 'doc/*'
-  - 'LICENSE'
 
-# Ignore security incidents with the SHA256 of the occurrence obtained at output or the secret itself
 matches-ignore:
-  - name:
+  - SOME_SECRET
+  - name: foo
     match: 530e5a4a7ea00814db8845dd0cae5efaa4b974a3ce1c76d0384ba715248a5dc1
-  - name: credentials
-    match: MY_TEST_CREDENTIAL
 
-show-secrets: false # default: false
+banlisted-detectors:
+  - Generic Password
+```
 
-# Set to true if the desired exit code for the CLI is always 0,
-# otherwise the exit code will be 1 if incidents are found.
-# the environment variable GITGUARDIAN_EXIT_ZERO=true can also be used toggle this behaviour.
-exit-zero: false # default: false
+And here is the equivalent v2 file:
```

Comment:
Since we already have a way to serialize/deserialize configs, did we think about having an automatic generation of v2 config files from v1 config files?